packages/application-shell/src/components/application-shell/application-shell.spec.js

```diff
@@ -692,12 +692,13 @@ describe('navbar menu links interactions', () => {
     rendered,
     { mainMenuLabel, mainSubmenuLabel }
   ) {
-    // Get the nav container, to narrow down the search area
-    const leftNavContainer = await rendered.findByTestId('left-navigation');
-    const navbarRendered = within(leftNavContainer);
+    // Wait for the nav container
+    await rendered.findByTestId('left-navigation');
 
     // Check the relationships between the menu items of a group
-    const menuTitle = await navbarRendered.findByText(mainMenuLabel.value);
+    const menuTitle = await within(
+      await rendered.findByTestId('left-navigation')
```

Comment:
This is odd. The tests fail cause the return of the `within`-call is bound to a subdom (only under JSDOM v16) which contains a menu which is still loading. According to the classname and content. While the debug value or rendered clearly outlines the opposite.

I then tried to use waitForElementToBeRemoved to await the loading to be removed. However, rtl rightfully tells me that it was never present in the first place. Hence, it being an invalid await.

I then inlined awaiting the left navigation which into a within in place. This then fixes the tests. It's the only way I got things to pass.

Reply:
P.S.: I am fine with this. However, I would also like the render to be a bit more helpful like `rendered.findByLeftNavigation`.